sympy/printing/latex.py

```diff
@@ -1721,6 +1720,19 @@ def _print_MatMul(self, expr):
         else:
             return ' '.join(map(parens, args))
 
+    def _print_Determinant(self, expr, exp=None):
+        if expr.args[0].is_MatrixExpr:
+            tex = r"\left|{%s}\right|" % self._print(expr.args[0])
+        else:
+            mat_delim_backup = self._settings['mat_delim']
+            self._settings['mat_delim'] = ''
+            tex = r"\left|{%s}\right|" % self._print(expr.args[0])
```

Comment:
I think you mentioned this on the issue, but why not just set the delimiter to `|` and recursively print `expr.args[0]`? The result would be the same but would be less convoluted.

Reply:
Main reason is that it was like that in the code.

The corresponding code would be:
```
            self._settings['mat_delim'] = '|'
            tex = self._print(expr.args[0])
```
and changing
```
        self._delim_dict = {'(': ')', '[': ']'}
```
to 
```
        self._delim_dict = {'(': ')', '[': ']', '|': '|'}
```
so to me it seemed about the same.

But I can definitely change it. One should probably check how block matrices print though... Things like 
```
X = MatrixSymbol('X', 2, 2)
m = Matrix(((1, 2), (3, 4)))
BlockMatrix(((OneMatrix(2, 2), X), (m, ZeroMatrix(2, 2))))
```
Will probably not print in a sensible way independent of solution...

(Right now:
```
⎡  𝟙     X⎤
⎢         ⎥
⎢⎡1  2⎤   ⎥
⎢⎢    ⎥  𝟘⎥
⎣⎣3  4⎦   ⎦
```
)